### PR TITLE
#2717 Hide Keyboard when Scrolling in OnlineLibraryFragment/PagerFragment

### LINKED ISSUE #2717: Hide Keyboard when Scrolling in OnlineLibraryFragment/PagerFragment
**Is your feature request related to a problem? Please describe.**
In  OnlineLibraryFragment  & PagerFragment  while scrolling the keyboard just stays there and takes so much space of the page
which in my opinion isn't a great experience, its better if we scroll and the keyboard goes away :)

**Describe the solution you'd like**
I will addOnScrollListener to the recycler view so everytime we scroll the keyboard gets hidden

**Describe alternatives you've considered**
if we use a touch listener, then the keyboard hiding code is called multiple times each second while scrolling, which is not efficient.

**Additional context**
Before my approach 

https://user-images.githubusercontent.com/75211982/137470164-0d79d9d4-6856-4956-88b2-d8fcdc72a653.mp4

After my approach 

https://user-images.githubusercontent.com/75211982/137470182-efae0cbd-4e29-42df-bff1-d60632b8363f.mp4

### PR DESCRIPTION
**Pull request by cdhiraj40**
Fixes #2717

**Screenshots** 

https://user-images.githubusercontent.com/75211982/137470182-efae0cbd-4e29-42df-bff1-d60632b8363f.mp4

### CODE REVIEW

**Comment by cdhiraj40** [comment]
@gouri-panda @kelson42 should i replace ``requireFragmentManager()`` with ``parentFragmentManager`` as ```'requireFragmentManager() is  deprecated ```

**Comment by gouri-panda** [comment]
Yes! Please avoid deprecated classes/methods.

**Comment by cdhiraj40** [comment]
> Yes! Please avoid deprecated classes/methods.

done :)

File: core/src/main/java/org/kiwix/kiwixmobile/core/utils/SimpleRecyclerViewScrollListener.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    onLayoutScrollListener(
+      recyclerView,
+      newState
+    ) // implement the callback by passing recyclerview and  newState
+    super.onScrollStateChanged(recyclerView, newState)
+  }
+}
+
+
+
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/utils/SimpleRecyclerViewScrollListener.kt:38]: You should remove these blank lines that's why static analyses are failing.

File: core/src/main/java/org/kiwix/kiwixmobile/core/utils/SimpleRecyclerViewScrollListener.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  private val onLayoutScrollListener: (RecyclerView, Int) -> Unit // here we are calling callback
+) :
+  RecyclerView.OnScrollListener() {
+  override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
+    super.onScrollStateChanged(recyclerView, newState)
+    onLayoutScrollListener(
+      recyclerView,
+      newState
+    ) // implement the callback by passing recyclerview and  newState
+    super.onScrollStateChanged(recyclerView, newState)
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/utils/SimpleRecyclerViewScrollListener.kt:33]: You are calling twice super method. Remove this one.

File: core/src/main/java/org/kiwix/kiwixmobile/core/utils/SimpleRecyclerViewScrollListener.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.utils
+
+import androidx.recyclerview.widget.RecyclerView
+
+class SimpleRecyclerViewScrollListener(
+  private val onLayoutScrollListener: (RecyclerView, Int) -> Unit // here we are calling callback
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/utils/SimpleRecyclerViewScrollListener.kt:24]: Don't need this comment but if you want to make a meaningful comment :)

**Review by gouri-panda — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/OnlineLibraryFragment.kt [hunk 4]
```diff
@@ -265,5 +274,5 @@ class OnlineLibraryFragment : BaseFragment(), FragmentActivityExtensions {
     .apply {
       onSelectAction = ::storeDeviceInPreferences
     }
-    .show(requireFragmentManager(), getString(R.string.pref_storage))
+    .show(parentFragmentManager, getString(R.string.pref_storage))
```
Comment (gouri-panda) [inline app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/OnlineLibraryFragment.kt:277]: It's good you wanna remove the deprecated method/s. but it's off-topic here. Create an issue for this and then change theire.

File: core/src/main/java/org/kiwix/kiwixmobile/core/utils/SimpleRecyclerViewScrollListener.kt (hunk 1 above)
Comment (cdhiraj40) [reply core/src/main/java/org/kiwix/kiwixmobile/core/utils/SimpleRecyclerViewScrollListener.kt]: done.

**Review by gouri-panda — APPROVED** [review]
@cdhiraj40 Thank you for the contribution 🎉

**Pull request merged**